x/concentrated-liquidity/keeper_test.go

```diff
@@ -313,3 +310,25 @@ func (s *KeeperTestSuite) validateListenerCallCount(
 func (s *KeeperTestSuite) setListenerMockOnConcentratedLiquidityKeeper() {
 	s.App.ConcentratedLiquidityKeeper.SetListenersUnsafe(types.NewConcentratedLiquidityListeners(&clmocks.ConcentratedLiquidityListenerMock{}))
 }
+
+// Crosses the tick and charges the fee on the global fee accumulator.
+// This mimics crossing an initialized tick during a swap and charging the fee on swap completion.
+func (s *KeeperTestSuite) crossTickAndChargeFee(poolId uint64, tickIndexToCross int64) {
+	// Cross the tick to update it.
+	_, err := s.App.ConcentratedLiquidityKeeper.CrossTick(s.Ctx, poolId, tickIndexToCross, DefaultFeeAccumCoins[0])
+	s.Require().NoError(err)
+	err = s.App.ConcentratedLiquidityKeeper.ChargeFee(s.Ctx, poolId, DefaultFeeAccumCoins[0])
+	s.Require().NoError(err)
+}
+
+func (s *KeeperTestSuite) validatePositionFeeGrowth(poolId uint64, positionId uint64, expectedUnclaimedRewards sdk.DecCoins) {
+	accum, err := s.App.ConcentratedLiquidityKeeper.GetFeeAccumulator(s.Ctx, poolId)
+	s.Require().NoError(err)
+	positionRecord, err := accum.GetPosition(cltypes.KeyFeePositionAccumulator(positionId))
+	s.Require().NoError(err)
+	if expectedUnclaimedRewards.IsZero() {
+		s.Require().Equal(expectedUnclaimedRewards, positionRecord.UnclaimedRewards)
+	} else {
+		s.Require().Equal(expectedUnclaimedRewards[0].Amount.Mul(DefaultLiquidityAmt), positionRecord.UnclaimedRewards.AmountOf(expectedUnclaimedRewards[0].Denom))
+	}
+}
```

Comment:
Should we just take in a DecCoin here? Is it possible for fee growth to contain more than just a single denom?

Reply:
Since fee is always charged on the input token, I believe it can be either of the two denoms in the pool

Reply:
So then unclaimedRewards could potentially have two indices right? And it looks like we are only using the first index here.

Reply:
Updated to validate both denoms if two are exptected: [e89cb65](https://github.com/osmosis-labs/osmosis/pull/4973/commits/e89cb6581c985df2de9f1c6562bfbaa4a5ecf8ef)